USWDS - Header: Alignment fixes

amyleadem opened the pull request:
# Summary
Updated the padding on `.usa-nav__submenu` elements so that they align with other header elements.

## Breaking change

This is not a breaking change.
However, we recommend that users confirm that the header elements in their project align as expected. 

## Related issue

Closes #5417
Closes #5604 

## Related pull requests

This PR is a combination of PRs #5418 and #5630.

Changelog PR

## Preview link

- [Header component - default](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/header-nav-link-alignment/?path=/story/components-header--default)
- [Header component - extended](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/header-nav-link-alignment/?path=/story/components-header--extended)
- [Header component - megamenu](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/header-nav-link-alignment/?path=/story/components-header--megamenu)
- [Header component - extended megamenu ](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/header-nav-link-alignment/?path=/story/components-header--extended-megamenu)

## Problem statement

Header navigation links do **not** align with other header elements. Standard variants align too far to the left and megamenu variants align too far to the right. 

From PR #5418:
> USWDS 2 had some padding in nav submenus so that the submenu items were visually aligned with the nav menu button itself (see https://github.com/uswds/uswds/blob/v2.13.3/src/stylesheets/components/_navigation.scss#L374). This padding was removed in USWDS 3, and the nav submenus look off. 

- **[Develop - default](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/?path=/story/components-header--default)**
    
    ![image](https://github.com/uswds/uswds/assets/93996430/4948c7f7-0efe-4e1d-a76e-77f0de007aad)
- **[Develop - extended](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/?path=/story/components-header--extended)**
    
    ![image](https://github.com/uswds/uswds/assets/93996430/fa27ae3d-7871-4375-b166-a5f6dc61e574)

- **[Develop - megamenu](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/?path=/story/components-header--megamenu)**
    
    ![image](https://github.com/uswds/uswds/assets/93996430/027722b0-0b1a-46b7-8951-c617b58fbdf9)
- **[Develop - extended megamenu](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/?path=/story/components-header--extended-megamenu)**

    ![image](https://github.com/uswds/uswds/assets/93996430/d11a3423-d53e-46ae-b765-5e3ac09847e9)

## Solution

- [Header component - default](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/header-nav-link-alignment/?path=/story/components-header--default)

    ![image](https://github.com/uswds/uswds/assets/93996430/71989199-4802-4b28-823b-aa1e652879fb)

- [Header component - extended](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/header-nav-link-alignment/?path=/story/components-header--extended)
   
    ![image](https://github.com/uswds/uswds/assets/93996430/1767c6b3-d172-4e19-b0f0-f52b10db3031)

- [Header component - megamenu](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/header-nav-link-alignment/?path=/story/components-header--megamenu)
    
    ![image](https://github.com/uswds/uswds/assets/93996430/87072f2f-6dd6-44c7-904e-d99559a73e57)

- [Header component - extended megamenu ](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/header-nav-link-alignment/?path=/story/components-header--extended-megamenu)
    - Some changes in overall submenu width to create right alignment
    
    ![image](https://github.com/uswds/uswds/assets/93996430/8f6e61df-9db6-49d9-8926-945b8d1465a2)

## Testing and review

- Open each variant of the header component and confirm:
    - All navigation links align to the left of the other header elements.
    - Megamenu link elements also align to the right of other header elements.
    - Mobile presentation is not affected. 
    - No visual regressions

mejiaj commented:
@amyleadem 

### Right alignment

> * I made an update to standardize right alignment on the extended megamenu variant in [0d65c1c](https://github.com/uswds/uswds/pull/5649/commits/0d65c1c8e610b9c291123c4e9a72cdd5d4fe8515). The content extended beyond the width of the navbar, and it would be great to get your confirmation that this was the correct choice.

Makes sense. I tried to keep the original PR as simple as possible to avoid regressions and to speed up review/testing. Thanks for making this fix.

### Narrow links in Megamenu

> * There is now increased left padding on link elements in the megamenu variants. This makes the link content a bit narrower, which makes me a bit concerned about increased risk of new line breaks (though this will also be true of the change in [0d65c1c](https://github.com/uswds/uswds/pull/5649/commits/0d65c1c8e610b9c291123c4e9a72cdd5d4fe8515)). Curious to hear your thoughts on this.

How much narrower? Is this a big impact? Otherwise makes sense to continue.

amyleadem commented:
@mejiaj The link elements are now about 16px narrower than before. I included some screenshots in the description to show the change. It should be easiest to see in the screenshot for Header component - megamenu. I believe this is happening because the [updated link padding](https://github.com/uswds/uswds/pull/5649/files#diff-49f6f5eebd50f191ed35f60fb9582fe43bcf3d5b1ef18d562cee9419ab4df0bcR401) is being universally applied to all variants. 

We could either narrow how that padding is being applied or add an override to the megamenu links. I'd expect it to be a quick one-line fix to revert the megamenu link styles. 

My preference to make the change to preserve the link width in megamenu variants to match the narrower padding in `develop`. Curious to hear what you think.

amyleadem commented:
@mejiaj @mahoneycm I added a padding override for megamenu submenu links in [7222ba5](https://github.com/uswds/uswds/pull/5649/commits/7222ba50a8d8d4e7602fbe9a89250c1f104e1b63) and I updated the related screenshots in the description. Let me know what you think!

mahoneycm reviewed (approved):
Lgtm! Thanks Amy!

- [x]  Standard dropdown links properly aligned to their parent dropdown text
- [x]  Megamenu links do not surpass search bar
- [x]  Megamenu maintains expected line breaks 
- [x]  Mobile presentation unaffected
- [x]  No unintentional visual regressions

mejiaj reviewed (approved):
Thanks for the changes.

I tested mobile and desktop, all variants. Both left/right alignment looks good.
